### DISCUSSION TITLE
Microphone array calibration

### DISCUSSION THREAD

**Discussion by shakibmehr** [acoular/acoular#339] [Q&A] [answered]
Why calibrate the array?
What tool should I calibrate with?
How do I do this with Acoular?

**Comment by adku1173** [comment]
Hi @shakibmehr, can you explain what kind of calibration you are referring to and what you want to achieve? Currently, Acoular supports per-channel calibration via the `Calib` class. This class allows to apply calibration factors to each channel of an array in the time domain. The factors are in [Pa/unit]. An example using this functionality can be found here: https://www.acoular.org/auto_examples/wind_tunnel_examples/example_airfoil_in_open_jet_cmf.html#sphx-glr-auto-examples-wind-tunnel-examples-example-airfoil-in-open-jet-cmf-py

**Reply by shakibmehr** [reply]
My question is exactly what is the need to calibrate the Mems microphone array for beamforming?

**Reply by artpelling** [reply, marked as answer]
Calibration is used to achieve the correct scaling of values and physical units. As `acoular` is a package for acoustic testing, the beamforming processing pipeline expects sound pressure [Pa]. If calibration is omitted, this can lead to unwanted behaviour if levels are involved, e.g. when using `acoular.L_p`, as the reference value would no longer be correct. `acoular` uses the reference sound pressure of `p0=2e-5` as usual in acoustics.
Furthermore, a calibration per channel can improve the computed results as variablities in the microphone sensitivities are accounted for.

**Reply by shakibmehr** [reply]
thanks for response dear
How can a MEMS microphone array be calibrated for beamforming? What is the exact method for performing this calibration?

**Reply by m-s-y** [reply]
> What is the exact method for performing this calibration?

seconded.

**Reply by shakibmehr** [reply]
Where was it sent?!
